### PR TITLE
When binding roles, the user list is changed to the drop-down selection mode （在绑定角色的时候用户列表改成下拉选中的模式) #3518

### PR DESCRIPTION
**Pull request by zzq1314zll**
在绑定角色的时候用户列表改成下拉选中的模式 #3518

### CODE REVIEW

**Comment by KomachiSion** [comment]
Please use nacos code style to reformat your change of java code
detail see src/style/codeStyle.md

File: config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java [hunk 1]
```diff
@@ -39,13 +40,13 @@
 @Conditional(value = ConditionOnEmbeddedStorage.class)
 @Component
 public class EmbeddedUserPersistServiceImpl implements UserPersistService {
-    
+
```
Comment (KomachiSion) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java:43]: Do not change indent please.
please check for all your changed files

Comment (KomachiSion) [reply config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java:43]: Github show the indent still be changed

Comment (zzq1314zll) [reply config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java:43]: I am changing

**Comment by zzq1314zll** [comment]
@ready to close@

File: config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java [hunk 2]
```diff
@@ -117,4 +118,11 @@ public User findUserByUsername(String username) {
         }
         return pageInfo;
     }
+
+    @Override
+    public List<User> findUserLikeUsername(String username) {
+        String sql = "SELECT username FROM users WHERE username like '%' ? '%'";
```
Comment (KomachiSion) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java:124]: Please use prepareStatement and `?` SQL to resolve SQL injection risk.

Just like `findUserByUsername` method

[the remark this answers is not in the document]
Comment (KomachiSion) [reply config/src/main/java/com/alibaba/nacos/config/server/auth/EmbeddedUserPersistServiceImpl.java:124]: I think the SQL will be handle to `SELECT username FROM users WHERE username like '%' 'xxx' '%';` if username is xxx

File: config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalUserPersistServiceImpl.java [hunk 3]
```diff
@@ -145,4 +146,11 @@ public User findUserByUsername(String username) {
             throw e;
         }
     }
+
+    @Override
+    public List<User> findUserLikeUsername(String username) {
+        String sql = "SELECT username,password FROM users WHERE username like '%' ? '%'";
```
Comment (KomachiSion) [inline config/src/main/java/com/alibaba/nacos/config/server/auth/ExternalUserPersistServiceImpl.java:152]: Only query username in Embedded, but username,password in there?

**Comment by zzq1314zll** [comment]
@KomachiSion  Don't close this ,I will optimize similar functions in other places tomorrow
不要关闭这个issues ,我明天把其他页面相似的功能加上

**Comment by KomachiSion** [comment]
> @KomachiSion Don't close this ,I will optimize similar functions in other places tomorrow
> 不要关闭这个issues ,我明天把其他页面相似的功能加上

One PR one change, If you want to do some change about other feature, please open new PR.

**Pull request merged**